docs/best-practices.md

```diff
@@ -0,0 +1,73 @@
+# Best Practices
+
+This page provides information on how to get the most out of Granite.Code. It covers best practices for getting accurate and helpful responses, as well as a range of supplementary features that are included in Granite.Code.
+
+Before you read this guide, you should be familiar with the main Granite.Code features, which are described in the [getting started tutorial](FIXME).
+
+## Ask effective questions
+
+Being able to chat with a local AI assistant is one of the main features offered by Granite.Code. The following guidelines are recommended for asking questions, in order to get the most useful responses:
+
+* Be specific and precise - avoid broad or vague questions
+* Break complex questions down into sections - you can use bullet points or sections
+* If you want the answer in a specific format, describe it
+* Ask follow-up questions to refine the response
+* Provide the right context for your question (see below)
+
+## Provide context for queries
+
+By default, when making requests, an AI model has no awareness of the objects or environment that the query is related to - each response is based on general information that is available to the model, rather than specific information about the project or current focus. As a result, by default, requests to the model will return general responses.
+
+Setting context allows you to provide additional information to the model about what you are asking about. Granite.Code allows a variety of different types of context to be specified for a given chat prompt:
+
+* Code snippets: sections of the code that you can include in your query
+* Codebase: to ask a question about an entire project of code
+* Files: to ask questions about specific files
+* Docs: to ask questions about a particular source of documentation
+* Git diffs: to ask questions about a change in version control
+* Folders: to ask questions about a directory of files
+
+To add any of these types of context to a query, just type an **@** symbol, and a list of available context types will be shown. You can also click the **@** button in the chat prompt.
+
+Examples of context-based queries you can ask include:
+
+* TODO
+* TODO
+* TODO
+
+## Ask about your codebase
+
+Codebase is a special type of context, which can be used to ask questions about an entire project. This can be useful for learning how a project works, or finding ways to improve a project.
+
+To ask a question about a codebase, you can either:
+
+* Include **@Codebase** in your chat query, or
+* Enter your query and click the **Ctrl⏎ @codebase** button, or
+* Enter your query and press **Ctrl+⏎**
```

Comment:
there's a bit of inconsistency here. sometimes you use `Ctrl+⏎` sometimes `Ctrl⏎` sometimes you say `⏎ Enter` sometimes, you say `Enter` and sometimes you say `⏎`. For Tab you just say `Tab` and not `↹` or `↹ Tab` (and we don't use ⌘ at all).  Not sure what the right answer is...